spinnaker/spinnaker_system/google_http_upsert_test.py

```diff
@@ -0,0 +1,891 @@
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# See testable_service/integration_test.py and spinnaker_testing/spinnaker.py
+# for more details.
+#
+# The Http(s) test will be used to (more) thoroughly test the GCP L7 load
+# balancer Upsert and Delete commands, and will eventually be expanded to
+# include server group tests.
+
+# Standard python modules.
+from OpenSSL import crypto
+import copy
+import json
+import sys
+import time
+import uuid
+
+# citest modules.
+from citest.base import ExecutionContext
+import citest
+import citest.gcp_testing as gcp
+import citest.json_predicate as jp
+import citest.service_testing as st
+
+# Spinnaker modules.
+import spinnaker_testing as sk
+import spinnaker_testing.gate as gate
+
+HTTP_PROXY_PREFIX = 'target-http-proxy'
+HTTPS_PROXY_PREFIX = 'target-https-proxy'
+SCOPES = [gcp.COMPUTE_READ_WRITE_SCOPE]
+GCE_URL_PREFIX = 'https://www.googleapis.com/compute/v1/projects/'
```

Comment:
Couldnt you add a gcp_agent.getResourceUrl() method and use the URL from the get method from the discovery document in the gcp_compute_agent as I'd suggested offline?